src/sets.jl

```diff
@@ -1143,6 +1143,361 @@ struct AllDifferent <: AbstractVectorSet
+        n::Base.Integer,
+        partitions::Vector{Int},
+        set::Set{Int},
+    )
+        if any(p <= 0 for p in partitions)
+            throw(DimensionMismatch("Invalid partition dimension."))
+        end
+        return new(n, partitions, set)
+    end
+end
+
+dimension(s::CountAtLeast) = sum(s.partitions)
+
+function Base.:(==)(x::CountAtLeast, y::CountAtLeast)
+    return x.n == y.n && x.partitions == y.partitions && x.set == y.set
+end
+
+"""
+    CountGreaterThan(dimension::Int)
+
+The set ``\\{(c, y, x) \\in \\mathbb{Z}^{1+1+d}\\}`` such that `c` is strictly
+greater than the number of occurances of `y` in `x`.
+
+## Also known as
+
+This constraint is called `count_gt` in MiniZinc.
+
+## Example
+
+```julia
+model = Utilities.Model{Float64}()
+c, _ = add_constrained_variable(model, Integer())
+y, _ = add_constrained_variable(model, Integer())
+x = [add_constrained_variable(model, Integer())[1] for _ in 1:3]
+add_constraint(model, VectorOfVariables([c; y; x]), CountGreaterThan(5))
+```
+"""
+struct CountGreaterThan <: AbstractVectorSet
+    dimension::Int
+    function CountGreaterThan(dimension::Base.Integer)
+        if dimension < 2
+            throw(
+                DimensionMismatch(
+                    "Dimension of CountGreaterThan must be >= 2.",
+                ),
+            )
+        end
+        return new(dimension)
+    end
+end
+
+"""
+    BinPacking(c::T, w::Vector{T}) where {T}
+
+The set ``\\{x \\in \\mathbb{R}^d\\}`` such that each item `i` in `1:d` of
+weight `w[i]` is put into bin `x[i]`, and the total weight of each bin does not
+exceed `c`.
+
+There are additional assumptions that the capacity, `c`, and the weights, `w`,
+must all be non-negative.
+
+## Also known as
+
+This constraint is called `bin_packing` in MiniZinc.
+
+## Example
+
+```julia
+model = Utilities.Model{Float64}()
+bins = add_variables(model, 5)
+weights = [1, 1, 2, 2, 3]
+add_constraint.(model, bins, MOI.Integer())
+add_constraint.(model, bins, MOI.Interval(1, 3))
+add_constraint(model, VectorOfVariables(bins), BinPacking(3, weights))
+```
+"""
+struct BinPacking{T} <: AbstractVectorSet
+    capacity::T
+    weights::Vector{T}
+    function BinPacking(capacity::T, weights::Vector{T}) where {T}
+        if capacity < zero(T)
+            throw(DomainError(capacity, "capacity must be non-negative"))
+        end
+        if any(w -> w < zero(T), weights)
+            throw(DomainError(weights, "weights must be non-negative"))
+        end
+        return new{T}(capacity, weights)
+    end
+end
+
+dimension(set::BinPacking) = length(set.weights)
+
+function Base.copy(set::BinPacking{T}) where {T}
+    return BinPacking(set.capacity, copy(set.weights))
+end
+
+function Base.:(==)(x::BinPacking{T}, y::BinPacking{T}) where {T}
+    return x.capacity == y.capacity && x.weights == y.weights
+end
+
+"""
+    Cumulative(dimension::Int)
+
+The set ``\\{(s, d, r, b) \\in \\mathbb{Z}^{length(s)+length(d)+length(r)+1}\\}``
```

Comment:
```suggestion
The set ``\\{(s, d, r, b) \\in \\mathbb{R}^{length(s)+length(d)+length(r)+1}\\}``
```

Reply:
I would imagine that MIP formulations would support these to be any real numbers

Reply:
I think most users and solvers of constraint programming in MOI are going to use `Model{Int}`, so I guess this becomes a moot point.